feugy/atelier
# feat: adds svelte-kit support

### What's in there?

While migrating Tabulous from svelte-spa-router to  Svelte-kit, I found a couple of issues preventing atelier to be used:

- Vite does not allow loading the tests files (they are outside of Vite allowed list)
- Vite displays a warning about @atelier-wb/svelte not being ESM nor commonjs 
- Svelte-kit displays a warning about section's redundant `region` role (which we absolutely need for testing with JS-dom)

### How to test?

1. create a new svelte-kit project: `npm init svelte my-app`
1. enter the created folder: `cd my-app`
1. `npm i -D @atelier-wb/vite-plugin-atelier @atelier-wb/svelte`
1. in `svelte.config.js`, add the following:
   ```js
   import atelier from '@atelier-wb/vite-plugin-atelier'
   // in exported config
   vite: { plugins: [atelier()] }
   ``` 
1. add a tool file: `atelier/Counter.tools.svelte`:
   ```js
   <script>
     import { Tool } from '@atelier-wb/svelte'
     import Counter from '../src/lib/Counter.svelte'
   </script>
   
   <Tool name="Counter" component={Counter} />
   ```
1. run vite: `npm run dev`
1. browse to http://localhost:3000/atelier
   > you should see Atelier's UI with a single tool loaded for the counter.
